File: Moxion/Domain/Kinematic/MotionData.cs
using Moxion.Abstractions;
using Moxion.Common.Enumerations;
using Moxion.Common.Values;
using Moxion.Common.Values.Derived;

namespace Moxion.Domain.Kinematic;

public readonly record struct MotionData(
  Time Time,
  Position Position,
  Velocity Velocity,
  Acceleration Acceleration,
  Jerk Jerk,
  MotionPhase Phase
) : IValueObject<MotionData>;

File: Moxion.Presentation.Web/ClientApp/src/models/kinematics/MotionData.ts
import MotionPhase from "../enumerations/MotionPhase.ts";

class MotionData {
    time: number
    position: number
    velocity: number
    acceleration: number
    jerk: number
    phase: MotionPhase

    constructor(
        time?: number,
        position?: number,
        velocity?: number,
        acceleration?: number,
        jerk?: number,
        phase?: MotionPhase
    ) {
        this.time = time ?? 0
        this.position = position ?? 0
        this.velocity = velocity ?? 0
        this.acceleration = acceleration ?? 0
        this.jerk = jerk ?? 0
        this.phase = phase ?? 'None'
    }
}

export default MotionData